data/blog/battery-life.mdx

```diff
@@ -0,0 +1,86 @@
+---
+title: 'Battery life instrumentation'
+date: '2023-04-25'
+tags:
+  - 'oss'
+  - 'mobile'
+authors: ['mihai']
+images: ['/images/battery-life/social.png']
+summary: 'Instrumentation that we’re adding to the Tailscale clients to have a scalable approach to battery life investigations.'
+---
+
+Battery consumption is one of the top areas where we’ve head that Tailscale needs to improve, via both [GitHub](https://github.com/tailscale/tailscale/issues/3363) and [social media](https://twitter.com/search?q=tailscale%20battery&src=typed_query&f=live). There’s always been [work on battery-related issues](https://github.com/tailscale/tailscale/issues?q=label%3Abattery+is%3Aclosed), but it’s not something that can ever be declared “done”:
+
+- Tailscale’s feature set evolves — using the [NextDNS integration](https://tailscale.com/blog/nextdns/) also means that Tailscale is now responsible for all DNS requests issued by a device.
```

Comment:
```suggestion
- Tailscale’s feature set evolves. For example, using the [NextDNS integration](https://tailscale.com/blog/nextdns/) means that Tailscale is now responsible for all DNS requests issued by a device.
```
Not sure I get this - weren't we already responsible with MagicDNS? Is the point that it's routing more DNS queries off the device?

Reply:
Good point. Though MagicDNS only went GA a few months ago, and was not enabled by default before then. Changed it to incorporate that too.